warehouse/query-core/src/main/java/datawave/query/util/sortedmap/HdfsBackedSortedMap.java

```diff
@@ -0,0 +1,302 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FsStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.log4j.Logger;
+
+import datawave.query.iterator.ivarator.IvaratorCacheDir;
+import datawave.query.iterator.ivarator.IvaratorCacheDirConfig;
+import datawave.query.util.sortedset.FileSortedSet;
+
+/**
+ * Sorted map backed by HDFS
+ *
+ * @param <K>
+ *            key of the map
+ * @param <V>
+ *            value of the map
+ */
+public class HdfsBackedSortedMap<K,V> extends BufferedFileBackedSortedMap<K,V> {
+    private static final Logger log = Logger.getLogger(HdfsBackedSortedMap.class);
+    private static final String FILENAME_PREFIX = "SortedMapFile.";
+
+    public static class Builder<B extends Builder<B,K,V>,K,V> extends BufferedFileBackedSortedMap.Builder<B,K,V> {
+        private List<IvaratorCacheDir> ivaratorCacheDirs;
+        private String uniqueSubPath;
+        private FileSortedSet.PersistOptions persistOptions;
+
+        public Builder() {
+            // change the default buffer persist threshold
+            withBufferPersistThreshold(10000);
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        protected B self() {
+            return (B) this;
+        }
+
+        public B withIvaratorCacheDirs(List<IvaratorCacheDir> ivaratorCacheDirs) {
+            this.ivaratorCacheDirs = ivaratorCacheDirs;
+            return self();
+        }
+
+        public B withUniqueSubPath(String uniqueSubPath) {
+            this.uniqueSubPath = uniqueSubPath;
+            return self();
+        }
+
+        public B withPersistOptions(FileSortedSet.PersistOptions persistOptions) {
+            this.persistOptions = persistOptions;
+            return self();
+        }
+
+        public HdfsBackedSortedMap<?,?> build() throws IOException {
+            return new HdfsBackedSortedMap<>(this);
+        }
+    }
+
+    public static HdfsBackedSortedMap.Builder<?,?,?> builder() {
+        return new HdfsBackedSortedMap.Builder<>();
+    }
+
+    protected HdfsBackedSortedMap(HdfsBackedSortedMap<K,V> other) {
+        super(other);
+    }
+
+    protected HdfsBackedSortedMap(Builder builder) throws IOException {
+        super(builder);
+        this.handlerFactories = createFileHandlerFactories(builder.ivaratorCacheDirs, builder.uniqueSubPath, builder.persistOptions);
+        // for each of the handler factories, check to see if there are any existing files we should load
+        for (SortedMapFileHandlerFactory handlerFactory : handlerFactories) {
+            // Note: All of the file handler factories created by 'createFileHandlerFactories' are SortedMapHdfsFileHandlerFactories
+            if (handlerFactory instanceof SortedMapHdfsFileHandlerFactory) {
```

Comment:
if this is so can this just be typed to SortedMapHdfsFileHandlerFactory in the field to simplify the code?

Reply:
yeah, that kinda looks ugly.  Trying to figure out why I did that.

Reply:
Updated to use the correct type.